### PR TITLE
Add typing-inspect.

### PR DESCRIPTION
**Pull request by tarcisioe**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by tarcisioe** [comment]
@conda-forge/staged-recipes Is this good to go? It passes all checks :)

File: recipes/typing-inspect/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - mypy_extensions >=0.3.0
+
+test:
+  imports:
+    - typing_inspect
+
+about:
+  home: http://github.com/ilevkivskyi/typing_inspect
+  license: MIT
+  license_family: MIT
```
Comment (chrisburr) [inline recipes/typing-inspect/meta.yaml]: The licence file isn't packaged (which is a requirement of the MIT licence). If it's included in the tarball you can specify it's name like this:
```suggestion
  license_file: LICENCE
```
Else you can copy the licence from the upstream repository and include it in the recipe directory (and then specify it's name like above).

Comment (chrisburr) [reply recipes/typing-inspect/meta.yaml]: Other than that it looks good to go 👍

Comment (tarcisioe) [reply recipes/typing-inspect/meta.yaml]: The file is not present either in the tarball or in the upstream repository. Is it reasonable for me to add the default MIT license file with the package author's name and current year?

Comment (chrisburr) [reply recipes/typing-inspect/meta.yaml]: That seems reasonable to me, though the year should probably be `2017-2019` i.e. the first commit.

Can you also make a PR upstream to add a copy of the licence and include it in the `MANIFEST.in` file?

Comment (tarcisioe) [reply recipes/typing-inspect/meta.yaml]: Yes, was going to do that as well. It is here already.

Comment (tarcisioe) [reply recipes/typing-inspect/meta.yaml]: Upstream PR: https://github.com/ilevkivskyi/typing_inspect/pull/33

**Pull request merged**